scripts/validate-gentx.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
+REGEN_HOME="/tmp/regen$(date +%s)"
+RANDOM_KEY="randomregenvalidatorkey"
+CHAIN_ID=regen-1
+
+GENTX_FILE=$(find ./$CHAIN_ID/gentxs -iname "*.json")
+LEN_GENTX=$(echo ${#GENTX_FILE})
+
+# Gentx Start date
+start="2021-03-31 15:00:00Z"
+# Compute the seconds since epoch for start date
+stTime=$(date --date="$start" +%s)
+
+# Gentx End date
+end="2021-04-06 15:00:00Z"
+# Compute the seconds since epoch for end date
+endTime=$(date --date="$end" +%s)
+
+# Current date
+current=$(date +%Y-%m-%d\ %H:%M:%S)
+# Compute the seconds since epoch for current date
+curTime=$(date --date="$current" +%s)
+
+if [[ $curTime < $stTime ]]; then
+    echo "start=$stTime:curent=$curTime:endTime=$endTime"
+    echo "Gentx submission is not open yet. Please close the PR and raise a new PR after 31-March-2021 15:00:00"
+    exit 0
+else
+    if [[ $curTime > $endTime ]]; then
+        echo "start=$stTime:curent=$curTime:endTime=$endTime"
+        echo "Gentx submission is closed"
+        exit 0
+    else
+        echo "Gentx is now open"
+        echo "start=$stTime:curent=$curTime:endTime=$endTime"
+    fi
+fi
+
+if [ $LEN_GENTX -eq 0 ]; then
+    echo "No new gentx file found."
+else
+    set -e
+
+    echo "GentxFile::::"
+    echo $GENTX_FILE
+
+    echo "...........Init Regen.............."
+
+    git clone https://github.com/regen-network/regen-ledger
+    cd regen-ledger
+    git checkout v1.0.0
+    make build
+    chmod +x ./build/regen
+
+    ./build/regen keys add $RANDOM_KEY --keyring-backend test --home $REGEN_HOME
+
+    ./build/regen init --chain-id $CHAIN_ID validator --home $REGEN_HOME
+
+    echo "..........Fetching genesis......."
+    rm -rf $REGEN_HOME/config/genesis.json
+    curl -s https://raw.githubusercontent.com/regen-network/mainnet/master/$CHAIN_ID/prelaunch-genesis.json >$REGEN_HOME/config/genesis.json
+
+    sed -i '/genesis_time/c\   \"genesis_time\" : \"2021-03-29T00:00:00Z\",' $REGEN_HOME/config/genesis.json
+
+    GENACC=$(cat ../$GENTX_FILE | sed -n 's|.*"delegator_address":"\([^"]*\)".*|\1|p')
+    amountquery=$(jq -r '.body.messages[0].value.amount' ../$GENTX_FILE)
```

Comment:
Is this intended to be used for anything?

We've spoken about having a max-submission requirement for GenTx. I'm waiting for final agreement on what that number is, but i'm assuming thats what this is for?

I'll let you know as soon as we have clarity on it.

Reply:
Yes, wanted to validate the max amount. Let me know if we have a number on this so we should update the instructions as well

Reply:
Number should be 50k REGEN